numpy/random/mtrand/mtrand.pyx

```diff
@@ -620,13 +620,63 @@ def _rand_bool(low, high, size, rngstate):
         return array
 
 
-def _rand_int8(low, high, size, rngstate):
+def _rand_bool(low, high, size, rngstate):
     """
-    _rand_int8(low, high, size, rngstate)
+    _rand_bool(low, high, size, rngstate)
 
     See `_rand_int32` for documentation, only the return type changes.
 
     """
+    cdef npy_bool off, rng, buf
+    cdef ndarray array "arrayObject"
+    cdef npy_bool *array_data
+    cdef npy_bool *high_data
+    cdef npy_bool *low_data
+    cdef broadcast multi
+    cdef npy_intp i
+
+    cdef ndarray lo = <ndarray>low
+    cdef ndarray hi = <ndarray>high
+    cdef rk_state *state = <rk_state *>NpyCapsule_AsVoidPtr(rngstate)
+
+    if size is None:
+        multi = <broadcast>PyArray_MultiIterNew(2, <void *>lo, <void *>hi)
```

Comment:
Sure thing.  Once I have a working implementation down, I'll run ```timeit``` with the C API calls, and then do a similar thing with the more "Pythonic" calls.